src/openssl_wrap.cpp

```diff
@@ -1323,5 +1323,29 @@ std::vector<uint8_t> _EC_KEY_key2buf(const EVP_PKEY* evp, point_conversion_form_
     return result;
 }
 
+
+std::vector<uint8_t> _ECDH_compute_key(const EVP_PKEY *pub_key, const EVP_PKEY *ecdh,
```

Comment:
> Users of the OpenSSL library are expected to normally use the EVP method for working with Elliptic Curve Diffie Hellman as described above and on the EVP Key Agreement page. The EVP API is implemented by a lower level ECDH API. In some circumstances, expert users may need to use the low level API. *This is not recommended for most users.*

Am I correct in assuming that we're using the lower level API for the KDF support?

Reply:
Hm, you don't seem to be using the KDF support at all. In that case, I'd prefer if we used the higher-level API, as recommended by the OpenSSL wiki.

Reply:
Done.
I've just don't have a good name for it and think about continue to use ECDH_compute_key.